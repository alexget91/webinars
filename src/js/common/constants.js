const PAGE_URL_PARAMETER = `page`;

const CardLabel = {
  BASIC: `basic`,
  ADVANCED: `advanced`,
};

export {PAGE_URL_PARAMETER, CardLabel};
