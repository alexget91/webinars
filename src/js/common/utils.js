const getBase64 = (file, onLoad) => {
  const reader = new FileReader();
  reader.readAsDataURL(file);
  reader.onload = () => {
    onLoad(reader.result);
  };
  reader.onerror = (error) => {
    console.error('Error: ', error);
  };
};

export {getBase64};
