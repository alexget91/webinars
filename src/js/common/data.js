import {CardLabel} from "./constants";

const catalogCards = [
  {
    id: 1,
    title: `HTTPS & SSL Does Not Mean You Have a Secure Website`,
    text: `Having an SSL certificate does not mean you have a secure website, and with the new European GDPR regulations`,
    imageSrc: `pictures/article1.jpg`,
  },
  {
    id: 2,
    title: `SEMrush 101`,
    text: `In this blog post, I share how to use three easily obtainable PPC reports to help improve your organic search performance.`,
    imageSrc: `pictures/article2.jpg`,
  },
  {
    id: 3,
    title: `SEMrush Real-Time Site Analysis of VisionDirect`,
    text: `Learn more about SEMrush`,
    imageSrc: `pictures/article3.jpg`,
    label: CardLabel.ADVANCED,
  },
  {
    id: 4,
    title: `SEO Surgery India`,
    text: `I couldn’t be happier to share with you data and insights from Ghergich & Co.’s latest partnership with SEMrush.`,
    imageSrc: `pictures/article2.jpg`,
    label: CardLabel.BASIC,
  },
  {
    id: 5,
    title: `HTTPS & SSL Does Not Mean You Have a Secure Website`,
    text: `Having an SSL certificate does not mean you have a secure website, and with the new European GDPR regulations`,
    imageSrc: `pictures/article1.jpg`,
  },
  {
    id: 6,
    title: `SEO Surgery India`,
    text: `I couldn’t be happier to share with you data and insights from Ghergich & Co.’s latest partnership with SEMrush.`,
    imageSrc: `pictures/article2.jpg`,
    label: CardLabel.BASIC,
  },
  {
    id: 7,
    title: `SEMrush 101`,
    text: `In this blog post, I share how to use three easily obtainable PPC reports to help improve your organic search performance.`,
    imageSrc: `pictures/article2.jpg`,
  },
];

export default catalogCards;
