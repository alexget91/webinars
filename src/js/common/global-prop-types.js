import PropTypes from "prop-types";
import {CardLabel} from "./constants";

const cardLabel = PropTypes.oneOf(Object.values(CardLabel));

const cardData = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  imageSrc: PropTypes.string.isRequired,
  label: cardLabel,
};

const cardFull = Object.assign({}, cardData, {
  url: PropTypes.string.isRequired,
  cssClass: PropTypes.string,
});

const bannerData = {
  title: PropTypes.string,
  srcBig: PropTypes.string.isRequired,
  srcSmall: PropTypes.string.isRequired,
};

const bannerFull = Object.assign({}, bannerData, {
  url: PropTypes.string.isRequired,
  cssClass: PropTypes.string,
});

const cardsArray = PropTypes.arrayOf(
  PropTypes.oneOfType([
    PropTypes.exact(cardData),
    PropTypes.exact(bannerData)
  ])
);

export {cardLabel, cardData, cardFull, bannerData, bannerFull, cardsArray};
