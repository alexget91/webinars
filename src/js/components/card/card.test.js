import React from "react";
import renderer from "react-test-renderer";
import {MemoryRouter} from "react-router-dom";
import Card from "./card";
import {CardLabel} from "../../common/constants";

it(`Card correctly renders after relaunch`, () => {
  const tree = renderer
    .create(<MemoryRouter>
      <Card
        id={0}
        title={`Title`}
        description={`description`}
        imageSrc={`imageSrc`}
        label={CardLabel.ADVANCED}
        url={`url`}
        cssClass={`cssClass`}
      />
    </MemoryRouter>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
