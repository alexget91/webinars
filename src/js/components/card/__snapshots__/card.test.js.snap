// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card correctly renders after relaunch 1`] = `
<article
  className="card cssClass"
>
  <a
    href="/url"
    onClick={[Function]}
  >
    <img
      alt="Title"
      className="card__image"
      src="imageSrc"
    />
  </a>
  <div
    className="card__content"
  >
    <h2
      className="card__title"
    >
      <a
        href="/url"
        onClick={[Function]}
      >
        Title
      </a>
    </h2>
    <p
      className="card__text"
    >
      description
    </p>
    <div
      className="card__footer"
    >
      <div
        className="label"
      >
        advanced
      </div>
    </div>
  </div>
</article>
`;
